api/org/commcare/api/persistence/SqliteIndexedStorageUtility.java

```diff
@@ -37,12 +37,23 @@
     private final Class<T> prototype;
     private final String tableName;
     private final String sandboxId;
+    private final File databaseFolder;
 
-    public SqliteIndexedStorageUtility(Class<T> prototype, String sandboxId, String tableName) {
+    public SqliteIndexedStorageUtility(Class<T> prototype, String sandboxId, String tableName){
+        this(prototype, sandboxId, tableName, null);
+    }
+
+    public SqliteIndexedStorageUtility(Class<T> prototype, String sandboxId, String tableName, String databasePath) {
         this.tableName = tableName;
         this.sandboxId = sandboxId;
         this.prototype = prototype;
 
+        if(databasePath == null){
+            databaseFolder = new File(DATABASE_FOLDER);
```

Comment:
Is it possible to pass this in from outside such that there is never a dependency between two different classes "Agreeing" on the default? It sketches me out in so many places we check for null and set to the default if so. Feels like it would be more consistent and reliable if all of these classes required the path location, and there was just one central place that set what it was if one was not provided.